invalid narrowing conversion from "int" to "unsigned long"

When building with intel's oneapi compiler `intel-oneapi-compilers@2022.1.0` I get a compiler error: 

```
/home/tdm39/nextsimdg/core/src/RectGridIO.cpp(92): error #3291: invalid narrowing conversion from "int" to "unsigned long"
      const std::vector<size_t> size = { metadata.getLocalExtentY(), metadata.getLocalExtentX() };
``` 

Also newer gnu compilers `gcc >= 16` will issue a warning:

```
/home/melt/sync/cambridge/projects/current/sasip/nextsimdg/core/src/RectGridIO.cpp: In member function ‘virtual
 Nextsim::ModelState Nextsim::RectGridIO::getModelState(const std::string&)’:
 /home/melt/sync/cambridge/projects/current/sasip/nextsimdg/core/src/RectGridIO.cpp:91:65: warning: narrowing conversion of ‘(&
 metadata)->Nextsim::ModelMetadata::getLocalCornerY()’ from ‘int’ to ‘long unsigned int’ [-Wnarrowing]
    91 |     const std::vector<size_t> start = { metadata.getLocalCornerY(), metadata.getLocalCornerX() };
       |                                         ~~~~~~~~~~~~~~~~~~~~~~~~^~
```  

The integers are always positive, so it should be fine to cast them to size_t.

--- a/core/src/RectGridIO.cpp
+++ b/core/src/RectGridIO.cpp
@@ -88,8 +88,10 @@ ModelState RectGridIO::getModelState(const std::string& filePath)
     // Set the origins and extensions for reading 2D data based
     // on MPI decomposition
     const auto& metadata = ModelMetadata::getInstance();
-    const std::vector<size_t> start = { metadata.getLocalCornerY(), metadata.getLocalCornerX() };
-    const std::vector<size_t> size = { metadata.getLocalExtentY(), metadata.getLocalExtentX() };
+    const std::vector<size_t> start = { static_cast<size_t>(metadata.getLocalCornerY()),
+        static_cast<size_t>(metadata.getLocalCornerX()) };
+    const std::vector<size_t> size = { static_cast<size_t>(metadata.getLocalExtentY()),
+        static_cast<size_t>(metadata.getLocalExtentX()) };
 #else
     const std::vector<size_t> start = { 0, 0 };
     const std::vector<size_t> size(ModelArray::dimensions(ModelArray::Type::H).rbegin(),

